File: python/consecutive-characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        max_power = 1
        current_power = 1
        c = s[0]
        for i in range(1, len(s)):
            if c == s[i]:
                current_power += 1
                max_power = max(max_power, current_power)
            else:
                c = s[i]
                current_power = 1
        return max(max_power, current_power)

File: rust/consecutive-characters.rs
use std::cmp;

impl Solution {
    pub fn max_power(s: String) -> i32 {
        let mut current = 'a';
        let mut current_power = 0;
        let mut max_power = 0;
        for c in s.chars() {
            if current == c {
                current_power += 1;
                max_power = cmp::max(current_power, max_power);
            } else {
                current = c;
                current_power = 1;
            }
        }
        return cmp::max(current_power, max_power);
    }
}
